src/System.Private.Xml/tests/XmlResolver/System.Xml.XmlResolver.Tests/XmlPreloadedResolverGetEntity.cs

```diff
@@ -81,17 +107,71 @@ public void XmlResolverGetEntityAsyncWithInvalidData()
+            xmlResolver.Add(new Uri("-//W3C//DTD FAKE 1.0 Not Real//EN", UriKind.RelativeOrAbsolute), inpData);
+            Task<object> output = xmlResolver.GetEntityAsync(new Uri("-//W3C//DTD FAKE 1.0 Not Real//EN", 
                 UriKind.RelativeOrAbsolute), null, typeof(Stream));
             var result = new byte[inpData.Length];
             (output.Result as Stream).Read(result, 0, result.Length);
             Assert.Equal(inpData, result);
         }
-                
+
+        [Fact]
+        public async Task XmlResolverGetKnownEntityAsync()
+        {
+            var xmlResolver = new XmlPreloadedResolver(XmlKnownDtds.All);
+            foreach (var dtdInfo in GetKnownDtds())
+            {
+                string expectedContent = NormalizeContent(File.ReadAllText(dtdInfo.SourcePath));
+
+                var uri = new Uri(dtdInfo.PublicId, UriKind.RelativeOrAbsolute);
+                Stream streamResult = await xmlResolver.GetEntityAsync(uri, null, null) as Stream;
+                Assert.NotNull(streamResult);
+                byte[] data = new byte[streamResult.Length];
+                streamResult.Read(data, 0, Convert.ToInt32(streamResult.Length));
+                Assert.Equal(expectedContent, NormalizeContent(Encoding.ASCII.GetString(data)));
+
+                uri = new Uri(dtdInfo.SystemId, UriKind.RelativeOrAbsolute);
+                streamResult = await xmlResolver.GetEntityAsync(uri, null, null) as Stream;
+                Assert.NotNull(streamResult);
+                data = new byte[streamResult.Length];
+                streamResult.Read(data, 0, Convert.ToInt32(streamResult.Length));
+                Assert.Equal(expectedContent, NormalizeContent(Encoding.ASCII.GetString(data)));
+            }
+        }
+
+        private static IEnumerable<(string PublicId, string SystemId, string SourcePath)> GetKnownDtds()
+        {
+            var xhtmlDtds = new (string PublicId, string RelativeId)[]
+            {
+                ("-//W3C//DTD XHTML 1.0 Strict//EN", "xhtml1-strict.dtd"),
+                ("-//W3C//DTD XHTML 1.0 Transitional//EN", "xhtml1-transitional.dtd"),
+                ("-//W3C//DTD XHTML 1.0 Frameset//EN", "xhtml1-frameset.dtd"),
+                ("-//W3C//ENTITIES Latin 1 for XHTML//EN", "xhtml-lat1.ent"),
+                ("-//W3C//ENTITIES Symbols for XHTML//EN", "xhtml-symbol.ent"),
+                ("-//W3C//ENTITIES Special for XHTML//EN", "xhtml-special.ent"),
+            };
+            var rssDtds = new (string PublicId, string RelativeId)[]
+            {
+                ("-//Netscape Communications//DTD RSS 0.91//EN", "rss-0.91.dtd"),
+            };
+
+            string dtdFolderRoot = @"Utils\DTDs\";
+
+            return Enumerable.Concat(
+                GetKnownDtds(xhtmlDtds, "http://www.w3.org/TR/xhtml1/DTD/", Path.Combine(dtdFolderRoot, @"XHTML10\no_comments\")),
```

Comment:
The backslashes in the path are presumably reason for failures. Just split it app into seperate args to `Path.Combine` (everywhere)

Reply:
Done. Thanks!